instrumentation/oppia_instrumentation_test.bzl

```diff
@@ -0,0 +1,40 @@
+"""
+Instrumentation macros to setup up end-to-end tests.
+"""
+
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
+
+def oppia_instrumentation_test(
+        name,
+        srcs,
+        deps):
+    """
+    Creates library, binary and a instrumentation test for each test suite.
+
+    Args:
+        name: str. The class name of the Test suite or the test file.
+        srcs: list of str. List of test files under instrumentation module.
+        deps: list of str. The list of dependencies needed to build and run the library.
+    """
+    kt_android_library(
+        name = name + "_lib",
+        testonly = True,
+        srcs = srcs,
+        visibility = ["//:oppia_testing_visibility"],
+        deps = deps,
+    )
+
+    native.android_binary(
+        name = name + "Binary",
+        testonly = True,
+        custom_package = "org.oppia.android",
+        instruments = "//:oppia_test",
+        manifest = "src/javatest/AndroidManifest.xml",
+        deps = [":" + name + "_lib"],
+    )
+
+    native.android_instrumentation_test(
```

Comment:
I think removing this would be better as this is not supported yet.

Reply:
I'd rather make sure that are setting up the skeleton structure correctly rather than leaving out a key part. The Bazel structure should work fine, we just can't run the test yet (which is also why we need to disable the tests for CI, but we would need to do that anyway since E2E tests can't run in GitHub Actions).

Reply:
I don't think we need a todo here.
Maybe a comment showing that it doesn't work yet.

Maybe something like
`// #3617: Not supported yet.`

Reply:
If there's remaining work, a TODO should track it (especially since that hooks into our new static analysis automation around making sure TODO issues are kept open until the work is fully resolved).

Reply:
Done